### PR TITLE
Build Manylinux Wheels

### PR DESCRIPTION
**Pull request by phschiele**
Hi @akshayka,

as discussed in #1367, this PR uses `cibuildwheel` to build manylinux wheels.
I have taken some inspiration from your and @bstellato's implementations and everything seems to be working as expected.

The only issue I found was that we need to run the manylinux build on the `manylinux_2_24` image. This means that the wheels only work with distros based on Debian 9 or higher (e.g. Ubuntu 18.04+). When running with other flags here, I got the following error:

```
ABI because of the presence of too-recent versioned symbols. You'll need to compile the wheel on an older toolchain
```
I used [auditwheel-symbols](https://github.com/messense/auditwheel-symbols) to inspect which symbols are causing the issues. Below are the outputs I got for manylinux 2014 and 2010, respectively.

```
_cvxcore.cpython-38-x86_64-linux-gnu.so is not manylinux_2_17(aka manylinux2014) compliant because it links the following forbidden libraries:
libstdc++.so.6  offending symbols: _ZNSt12out_of_rangeC1EPKc@@GLIBCXX_3.4.21, _ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE9_M_appendEPKcm@@GLIBCXX_3.4.21, _ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEED1Ev@@GLIBCXX_3.4.21, _ZNSt11logic_errorC2ERKS_@@GLIBCXX_3.4.21, _ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE9_M_createERmm@@GLIBCXX_3.4.21, _ZNSt16invalid_argumentC1EPKc@@GLIBCXX_3.4.21, _ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE6appendEPKc@@GLIBCXX_3.4.21
libstdc++.so.6  offending symbols: __cxa_throw_bad_array_new_length@@CXXABI_1.3.8
```

```
_cvxcore.cpython-38-x86_64-linux-gnu.so is not manylinux_2_12(aka manylinux2010) compliant because it links the following forbidden libraries:
libstdc++.so.6  offending symbols: __cxa_throw_bad_array_new_length@@CXXABI_1.3.8
libstdc++.so.6  offending symbols: _ZNSt12out_of_rangeC1EPKc@@GLIBCXX_3.4.21, _ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE9_M_appendEPKcm@@GLIBCXX_3.4.21, _ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEED1Ev@@GLIBCXX_3.4.21, _ZNSt11logic_errorC2ERKS_@@GLIBCXX_3.4.21, _ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE9_M_createERmm@@GLIBCXX_3.4.21, _ZNSt16invalid_argumentC1EPKc@@GLIBCXX_3.4.21, _ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE6appendEPKc@@GLIBCXX_3.4.21
libc.so.6       offending symbols: memcpy@@GLIBC_2.14
```

### CODE REVIEW

File: continuous_integration/GitHub/deploy_script.sh [hunk 1]
```diff
@@ -1,29 +0,0 @@
-#!/bin/bash
-
-source activate testenv
-conda config --add channels conda-forge
-conda install --yes requests twine
-pip install readme_renderer
```
Comment (phschiele) [inline continuous_integration/GitHub/deploy_script.sh:6]: Was `readme_renderer` actually used? If so, we could add it back as a separate build step.

Comment (SteveDiamond) [reply continuous_integration/GitHub/deploy_script.sh:6]: I believe readme_renderer is being used as part of the upload build, though I couldn't tell you what for exactly.

Comment (rileyjmurray) [reply continuous_integration/GitHub/deploy_script.sh:6]: I tried to remove it before and was unable (some errors appeared, I think with the PyPI upload). The reason why it's installed via pip is that we can't get the right version from conda.

**Comment by rileyjmurray** [comment]
There isn't much here that I can intelligently comment on. @phschiele my one question is: has this been tested for correctness somehow? In the past when I've set up our CI or deployment tools, I've made deployments to "Test PyPI" and my own conda channel.

**Comment by phschiele** [comment]
> There isn't much here that I can intelligently comment on. @phschiele my one question is: has this been tested for correctness somehow? In the past when I've set up our CI or deployment tools, I've made deployments to "Test PyPI" and my own conda channel.

@rileyjmurray In terms of testing, I downloaded the artifacts from the actions tab (e.g. the latest run [here](https://github.com/cvxpy/cvxpy/actions/runs/860620649)) and verified that the Linux wheel works as expected.
I suppose it would be quite easy to upload every "master merge" to Test PyPI and only the releases to PyPI itself, what do you think?

**Comment by rileyjmurray** [comment]
@phschiele in principle I'd be a big fan of uploading all master merge's to Test PyPI. The one issue there is that PyPI (and I believe Test PyPI) requires new version numbers for every release. Since only a portion of master merge's get incremented version numbers, that seems to prevent uploads for each master merge. Do you see a simple way around that?

**Comment by phschiele** [comment]
@rileyjmurray You're right, seems like even on Test PyPI one cannot overwrite existing versions. 
I guess a workaround could be to simply append the git hash to the filenames for Test PyPI, but not sure if that's a great idea.

**Comment by rileyjmurray** [comment]
@phschiele in that case I think it's not worth fussing with Test PyPI. However per your edit, @SteveDiamond @akshayka we might want to look at using GH Actions to increment version numbers (personally I think that's something to consider separately from this PR.)

**Comment by akshayka** [comment]
This looks great! I like that this lets us easily build with OpenMP (which we should do, at some point in the future).

Eventually, we should test and build for Windows using Github Actions as well (@rileyjmurray, do you agree?). I believe all that will require is adding a new OS to the build matrix. But we can handle that in a future change.

**Review by akshayka — APPROVED** [review]

**Review by SteveDiamond — APPROVED** [review]

**Pull request merged**